app/dashboard/src/providers/FeatureFlagsProvider.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file
+ *
+ * Feature flags provider.
+ * Feature flags are used to enable or disable certain features in the application.
+ */
+import { useMount } from '#/hooks/mountHooks'
+import { useLocalStorage } from '#/providers/LocalStorageProvider'
+import LocalStorage from '#/utilities/LocalStorage'
+import type { ReactNode } from 'react'
+import { useEffect } from 'react'
+import { z } from 'zod'
+import { createStore, useStore } from 'zustand'
+
+declare module '#/utilities/LocalStorage' {
+  /**
+   * Local storage data structure.
+   */
+  interface LocalStorageData {
+    readonly featureFlags: z.infer<typeof FEATURE_FLAGS_SCHEMA>
+  }
+}
+
+export const FEATURE_FLAGS_SCHEMA = z.object({
+  enableMultitabs: z.boolean(),
+  enableAssetsTableBackgroundRefresh: z.boolean(),
+  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+  assetsTableBackgroundRefreshInterval: z.number().min(100),
+})
+
+LocalStorage.registerKey('featureFlags', { schema: FEATURE_FLAGS_SCHEMA })
+
+/**
+ * Feature flags store.
+ */
+export interface FeatureFlags {
+  readonly featureFlags: {
+    readonly enableMultitabs: boolean
+    readonly enableAssetsTableBackgroundRefresh: boolean
+    readonly assetsTableBackgroundRefreshInterval: number
+  }
+  readonly setFeatureFlags: <
+    Key extends keyof FeatureFlags['featureFlags'],
+    Value extends FeatureFlags['featureFlags'][Key],
+  >(
+    key: Key,
+    value: Value,
```

Comment:
optional: this could be `FeatureFlags['featureFlags'][Key]` and then we can avoid an unnecessary second type parameter